scripts/AMDC_Logger.py

```diff
@@ -265,13 +258,49 @@ def load(self, file):
         out = pd.read_csv(p, comment = '#', index_col = 't')
         
         return out
+    
+    def _create_log_var(self, name, samples_per_sec, var_type, manual_index = None):
+        
+        memory_addr = int(self.mapfile.address(name), 0)
+        
+        if memory_addr == 0:
+            raise Exception(f"ERROR: couldn't find memory address for '{name}'")
+            
+        else:
+            if name not in self.log_vars.keys(): #check if variable name already registered
+
+                if manual_index is None:
+                    idx = self.available_indices.pop() #extract next available index
+                else:
+                    #remove index from list of available indices
+                    loc = self.available_indices.index(manual_index)
+                    idx = self.available_indices.pop(loc)
+                
+                #create LogVar object and append it to logvars
+                LV = LogVar(name = name, index = idx, var_type = var_type, samples_per_sec = samples_per_sec, memory_addr = memory_addr)
+                self.log_vars[name] = LV
```

Comment:
Can we be consistent with commenting? I see sometimes you put comments after a line of code, and sometimes the comment is before a block of code. I prefer all comments to go before lines of interest since this keeps line length minimal.